mongo/integration/cmd_monitoring_helpers_test.go

```diff
@@ -293,8 +315,14 @@ func compareStartedEvent(mt *mtest.T, expectation *expectation, id0, id1 bson.Ra
 
 		// Keys that may be nil
 		if val.Type == bson.TypeNull {
-			assert.NotNil(mt, err, "expected key %q to be omitted but got %q", key, actualVal)
-			continue
+			// Expected value is BSON null. Expect the actual field to be omitted.
+			if err == bsoncore.ErrElementNotFound {
+				continue
+			} else if err != nil {
+				return makeMatchError(mt, expected.Command, evt.Command, "expected key %q to be omitted but got error: %v", key, err)
+			} else {
+				return makeMatchError(mt, expected.Command, evt.Command, "expected key %q to be omitted but got %q", key, actualVal)
+			}
```

Comment:
```suggestion
			// Expected value is BSON null. Expect the actual field to be omitted.
			if errors.Is(err, bsoncore.ErrElementNotFound) {
				continue
			}

			if err != nil {
				return makeMatchError(mt, expected.Command, evt.Command, "expected key %q to be omitted but got error: %v", key, err)
			}

			return makeMatchError(mt, expected.Command, evt.Command, "expected key %q to be omitted but got %q", key, actualVal)

```


			
This suggestion:

(1) removes the if/else branching
(2) uses the `errors` package to check if the error is `bsoncore.ErrElementNotFound`